components/grid/selection/rows.md

```diff
@@ -0,0 +1,276 @@
+slug: components/grid/selection/rows
+tags: telerik,blazor,grid,selection,rows
+previous_url: /components/grid/selection/single,/components/grid/selection/multiple
+position: 3
+---
+
+# Row Selection
+
+The Grid component offers support for single or multiple row selection. You can select a row with mouse click and through checkbox column. You can access the collection of selected rows, use this collection and manipulate it. You can follow and respond to the event of selection.
+
+In this article:
+
+* [Rows Selection Options](#rows-selection-options)
+	* [Click-Only Selection](#click-selection)
+	* [Checkbox Selection](#checkbox-selection)
+* [Selected Rows](#selected-rows)
+	* [Basics](#basics)
+	* [Selected Rows When Data Changes](#selected-rows-when-data-changes)
+	* [Selected Rows Equals Comparison](#selected-rows-equals-comparison)
+* [SelectedItemsChanged](#selecteditemschanged)
+	* [SelectedItemsChanged and Asynchronous Operations](#selecteditemschanged-and-asynchronous-operations)
+* [Row Selection and Other Grid Features](#row-selection-and-other-grid-features)
+    * [Selection with Editing Modes](#selection-with-editing-modes)
+    * [Selection in Grid with virtualized rows](#selection-in-grid-with-virtualized-rows)
+    * [Selection and Grid Paging](#selection-and-grid-paging)
+    * [Selection in Template](#selection-in-template)
+    * [Selection and Row Drag and Drop](#selection-and-row-drag-and-drop)
+
+## Rows Selection Options
+
+By default you can [click](#click-selection) on the row to select it. Additionally, you can have a [checkbox selection](#checkbox-selection). The checkbox selection allows the user to select the row through a dedicated on the row checkbox. You can combine click and checkbox selection or use only one selection option.
+
+You can use row selection with both [selection modes]({%slug components/grid/selection/overview%}#selection-mode)—single and multiple.
+
+### Click Selection
+
+By default, users can select rows by clicking anywhere in the row, except on the command buttons.
+
+To select multiple rows, hold down the `Ctrl` or `Shift` key to extend the selection:
+* Press and hold `Ctrl` and click the desired rows to select or deselect them.
+* Click on the starting row in a range of rows that you want to select, press and hold `Shift`, and click on the last row in the range. The first selected row is the starting point of the range and the last selected row is the end of the selection.
+
+If you release the `Ctrl` or the `Shift` keys and click to start new multiple selection, the previously selected rows will be deselected.
+
+>caption Click selection and single selection mode
+
+````CSHTML
+Click on one row to select it.
+
+<TelerikGrid Data=@GridData
+             SelectionMode="@GridSelectionMode.Single"
```

Comment:
It is not a real-world scenario to enable selection and not define `SelectedItems`.